src/modules/esl-animate/core/esl-animate-mixin.ts

```diff
@@ -0,0 +1,69 @@
+import {jsonAttr, ESLMixinElement} from '../../esl-mixin-element/core';
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {ready} from '../../esl-utils/decorators/ready';
+import {memoize} from '../../esl-utils/decorators/memoize';
+import {parseNumber} from '../../esl-utils/misc/format';
+
+import {ESLAnimateService} from './esl-animate-service';
+
+interface AnimateOptions {
+  /**
+   * Class(es) to add on viewport intersection
+   * @see ESLAnimateConfig.cls
+   */
+  cls: string;
+  /**
+   * Re-animate item after its getting hidden
+   * @see ESLAnimateConfig.repeat
+   */
+  repeat: boolean;
+  /**
+   * Intersection ratio to consider element as visible.
+   * Only 0.2 (20%), 0.4 (40%), 0.6 (60%), 0.8 (80%) values are allowed due to share of IntersectionObserver instance
+   * with a fixed set of thresholds defined.
+   */
+  ratio: string;
+}
+
+/**
+ * ESLAnimateMixin - custom mixin element for quick {@link ESLAnimateService} attaching
+ *
+ * Use example:
+ * `<div esl-animate>Content</div>`
+ * Supports aditional parameters^
+ * `<div esl-animate={repeat: true, ratio: 0.8, cls: 'in'}>Content</div>`
+ */
+@ExportNs('AnimateMixin')
+export class ESLAnimateMixin extends ESLMixinElement {
+  public static is = 'esl-animate';
+
+  @jsonAttr({name: ESLAnimateMixin.is})
+  public options: AnimateOptions;
```

Comment:
It seems that AnimateOptions should be partial here as well, as the consumer should not care to define all of them.
So in that case it seems to be the AnimateOptions type itself should not be restrictive and should be optional out of the box